### DISCUSSION TITLE
How to edit or list a note with note ID within a folder, without mention the folder?

### DISCUSSION THREAD

**Discussion by stella10x1** [zk-org/zk#310] [Help] [answered]
Hello, I have following szenario:

The zk directory has 2 folders, e. g. `foo` and `bar`. And a file, which is known to have the id `ab12`. Unfortunately it is not known in which folder the file with id `ab12` is placed.

With `zk list ab12` or `zk edit ab12` I can't find the file without mention the right folder. As I don't know the folder I need to test both (E. g. `zk list foo/ab12`). Which isn't that bad for two folders, but get annoying for more than one folder.

Inside of Neovim the links do work without mentioning the folder, therefore I'm wondering if there is a way to access the file on command line without mentioning the folder?

**Comment by mickael-menu** [comment]
Are you using `[[WikiLink]]` or `[markdown](link)` with Neovim? If `[[WikiLink]]`, that explains the discrepancy. `zk list` takes paths but a `[[WikiLink]]` is more flexible. They are not yet exposed in the `zk` CLI.

However for now, you could use this:

```sh
zk list `find . -name *ab12*`
```

or better with the following `zk` alias:

```toml
[alias]
find = 'zk list `find . -name "*$1*"` $@'
```

You can use (the filename must be the first argument):

```sh
zk find ab12 --interactive
```

**Reply by stella10x1** [reply, marked as answer]
Thanks for your quick reply! Unfortunately the solution have two bugs:

1. The `find .` searches in the current directory.  That isn't a problem if `$PWD` is the zk dir, but does not work from arbitrary positions in the file tree
2. When there is no match it will output every note in the `$ZK_NOTEBOOK_DIR`.

My solution for those problem is the following:

```toml
[alias]
find = 'zk list $(results=$(find $ZK_NOTEBOOK_DIR -iname "*${1}*"); [ -z $results ] && echo /dev/null || echo "$results") $@'
```

As this is a little long alias and the syntax from shell is unreadable for the unfamiliar reader,  I will provide a small explanation:

- `find $ZK_NOTEBOOK_DIR -iname "*${1}*"` searches for a given filename only in `$ZK_NOTEBOOK_DIR` and outputs the absolute paths. The search is saved in the variable `results`. *This addresses bug 1*
- `[ -z $results ] && echo /dev/null || echo "$results")` This is just an if-else statement as one-liner. When there are no results it will output /dev/null, which isn't a zk note and therefore results in no notes found (*addresses bug 2*). If there are results those are simply printed to stdin and subsequently interpreted as parameters for `zk list`.
